python/src/trezorlib/cardano.py

```diff
@@ -298,37 +300,46 @@ def _parse_tokens(tokens: Iterable[dict], is_mint: bool) -> List[messages.Cardan
     return result
 
 
-def _parse_address_parameters(
-    address_parameters: dict, error_message: str
+def parse_address_parameters(
+    address_parameters: dict,
 ) -> messages.CardanoAddressParametersType:
-    if "addressType" not in address_parameters:
-        raise ValueError(error_message)
-
-    payment_path = tools.parse_path(address_parameters.get("path", ""))
-    staking_path = tools.parse_path(address_parameters.get("stakingPath", ""))
-    staking_key_hash_bytes = parse_optional_bytes(
-        address_parameters.get("stakingKeyHash")
-    )
-    script_payment_hash = parse_optional_bytes(
-        address_parameters.get("scriptPaymentHash")
-    )
-    script_staking_hash = parse_optional_bytes(
-        address_parameters.get("scriptStakingHash")
-    )
+    staking_path = address_parameters.get("staking_path")
 
     return create_address_parameters(
-        messages.CardanoAddressType(address_parameters["addressType"]),
-        payment_path,
-        staking_path,
-        staking_key_hash_bytes,
-        address_parameters.get("blockIndex"),
-        address_parameters.get("txIndex"),
-        address_parameters.get("certificateIndex"),
-        script_payment_hash,
-        script_staking_hash,
+        messages.CardanoAddressType(address_parameters["address_type"]),
+        tools.parse_path(address_parameters["path"]),
+        tools.parse_path(staking_path) if staking_path is not None else None,
+        parse_optional_bytes(address_parameters.get("staking_key_hash")),
+        address_parameters.get("block_index"),
+        address_parameters.get("tx_index"),
+        address_parameters.get("certificate_index"),
+        parse_optional_bytes(address_parameters.get("script_payment_hash")),
+        parse_optional_bytes(address_parameters.get("script_staking_hash")),
     )
 
 
+def parse_optional_address_parameters(
+    address_parameters: Optional[dict],
+) -> Optional[messages.CardanoAddressParametersType]:
+    if address_parameters is None:
+        return None
+
+    return parse_address_parameters(address_parameters)
+
+
+def _parse_output_address_parameters(
+    output: dict, error_message: str
+) -> messages.CardanoAddressParametersType:
+    if "addressType" not in output:
+        raise ValueError(error_message)
+
+    address_parameters = tools.dict_from_camelcase(output)
+    address_parameters.setdefault("path", "")
+    address_parameters.setdefault("staking_path", "")
```

Comment:
I'm not quite sure what's the actual purpose of `_parse_output_address_parameters`, but if it's only converting the (arguably wrong) camel case expected on some places, I would reflect that in the name (instead of calling it "output"). On the other hand, maybe we could stick to the camel case address parameters also for message signing, like everywhere else?

I don't see a reason why the optionality of `path` and `staking_path` couldn't be handled in `parse_optional_address_parameters`.